rmw_fastrtps_shared_cpp/src/rmw_count.cpp

```diff
@@ -54,12 +55,24 @@ __rmw_count_publishers(
   WriterInfo * slave_target = impl->secondaryPubListener;
   slave_target->mapmutex.lock();
   *count = 0;
-  for (const auto & it : slave_target->topicNtypes) {
-    const auto topic_fqdn = _demangle_if_ros_topic(it.first);
-    if (topic_fqdn == topic_name) {
-      *count += it.second.size();
+
+  // Build the list of all possible topic FQDN
+  std::vector<std::string> topic_fqdns;
```

Comment:
Maybe a `set` would be the better data structure to later check if a specific item is in it. I would assume it performs better.

Reply:
A `std::set` and `std::map` have the same runtime complexity `O(logN)` (for finding items). http://www.cplusplus.com/reference/set/set/find/ and http://www.cplusplus.com/reference/map/map/find/

There is no real differences but if you would rather have me use a `std::set`, I can change that.

Reply:
The current patch uses `std::vector` which afaik has `O(N)`. So `O(logN)` for a `std::set` would be an improvement.

Reply:
Sorry, you are correct. 
Will fix this as well.

Reply:
Ok, I'm getting myself confused. 
For the `topic_fqdns` variable, using a vector is more efficient as we push_back (`O(1)`) all possible topic names then iterate over that and find through the list of existing ROS2 topics (`O(logN)`). So, changing to using a set for this datastructure simply does not make sense.